### DISCUSSION TITLE
What is the recommended approach to dealing with parquet with nested columns?

### DISCUSSION THREAD

**Discussion by seandavi** [duckdb/pg_duckdb#258] [Q&A] [answered]
I'm trying to read a parquet file with a nested column and receiving this error:

```
ERROR:  a column definition list is required for functions returning "record"
```

Three of the columns are nested. I've seen some databases that simply suggest excluding nested columns (i.e., selecting only non-nested columns), but that doesn't seem to work. 

Thanks for the fun extension. I'm looking forward to see where it goes!

**Comment by wuputah** [comment, marked as answer]
The error you are seeing is because you must provide an `as` clause containing types for any columns you use in your select query (or at least 1 column if doing something like `count(*)`):

```sql
select max(example)
from read_parquet(...)
as (example int);
```

As for nested columns, those are not supported yet. I think if you try to read those, you will get a different error.
